packages/instrumentation-amqplib/test/amqplib-promise.spec.ts

```diff
@@ -0,0 +1,514 @@
+                'reject called on message without requeue'
+            );
+            expectConsumeEndSpyStatus([EndOperation.Reject, EndOperation.Reject]);
+        });
+
+        it('closing channel should end all open spans on it', async () => {
+            lodash.times(1, () => channel.sendToQueue(queueName, Buffer.from(msgPayload)));
+
+            await new Promise<void>((resolve) =>
+                asyncConsume(channel, queueName, [
+                    async (msg) => {
+                        await channel.close();
+                        resolve();
+                    },
+                ])
+            );
+
+            expect(memoryExporter.getFinishedSpans().length).toBe(2);
+            expect(memoryExporter.getFinishedSpans()[1].status.code).toStrictEqual(SpanStatusCode.ERROR);
+            expect(memoryExporter.getFinishedSpans()[1].status.message).toStrictEqual('channel closed');
+            expectConsumeEndSpyStatus([EndOperation.ChannelClosed]);
+        });
+
+        it('error on channel should end all open spans on it', (done) => {
+            lodash.times(2, () => channel.sendToQueue(queueName, Buffer.from(msgPayload)));
+
+            channel.on('close', () => {
+                expect(memoryExporter.getFinishedSpans().length).toBe(4);
+                // second consume ended with valid ack, previous message not acked when channel is errored.
+                // since we first ack the second message, it appear first in the finished spans array
+                expect(memoryExporter.getFinishedSpans()[2].status.code).toStrictEqual(SpanStatusCode.UNSET);
+                expect(memoryExporter.getFinishedSpans()[3].status.code).toStrictEqual(SpanStatusCode.ERROR);
+                expect(memoryExporter.getFinishedSpans()[3].status.message).toStrictEqual('channel error');
+                expectConsumeEndSpyStatus([EndOperation.Ack, EndOperation.ChannelError]);
+                done();
+            });
+            asyncConsume(channel, queueName, [
+                null,
+                (msg) => {
+                    try {
+                        channel.ack(msg);
+                        // ack the same msg again, this is not valid and should close the channel
+                        channel.ack(msg);
+                    } catch {}
+                },
+            ]);
+        });
+
+        // what should we do in this case?
+        // can cause memory leak since the plugin saves a copy of the msg
+        it.skip('throw exception from consumer callback', async () => {
+            lodash.times(1, () => channel.sendToQueue(queueName, Buffer.from(msgPayload)));
+
+            try {
+                await asyncConsume(channel, queueName, [
+                    (msg) => {
+                        throw Error('error from unit test');
+                    },
+                ]);
+            } catch (err) {
+                // console.log(err);
```

Comment:
forgotten comment

Reply:
Yeah this test is marked `skip`. it's still under construction until we resolve this issue